### PR TITLE
fix: corrected var/# parsing in template

### PR DESCRIPTION
**Pull request by mhevery**

### CODE REVIEW

File: modules/angular2/src/change_detection/parser/parser.ts [hunk 1]
```diff
@@ -532,7 +532,11 @@ class _ParseAST {
       this.optionalCharacter($COLON);
       var name = null;
       var expression = null;
-      if (this.next !== EOF) {
```
Comment (vicb) [inline modules/angular2/src/change_detection/parser/parser.ts]: wouldn't this work by only removing the enclosing `if (!EOF) { ...}`

**Comment by vicb** [comment]
You need to run `clang-format`

File: modules/angular2/test/change_detection/parser/parser_spec.js [hunk 2]
```diff
@@ -550,10 +550,16 @@ export function main() {
...
+        expect(keyValues(bindings)).toEqual(['#i=\$implicit']);
 
         bindings = parseTemplateBindings("#i");
-        expect(keyValues(bindings)).toEqual(['#i']);
+        expect(keyValues(bindings)).toEqual(['#i=\$implicit']);
+
+        bindings = parseTemplateBindings("var a; var b");
+        expect(keyValues(bindings)).toEqual(['#a=\$implicit', '#b=\$implicit']);
+
+        bindings = parseTemplateBindings("var a; var b;");
```
Comment (vicb) [inline modules/angular2/test/change_detection/parser/parser_spec.js]: should be `#a; #b`

**Comment by vicb** [comment]
@mhevery Why does the parser needs to explicitly set `'\$implicit'` ?
- It doesn't look like a job for a parser - the consumer could interpret null as he likes,
- Not sure to understand what issue this solves.

File: modules/angular2/src/change_detection/parser/parser.ts (hunk 1 above)
Comment (mhevery) [inline modules/angular2/src/change_detection/parser/parser.ts]: Good idea, implemented

**Comment by mhevery** [comment]
The issues is that `var x;` parses correctly, but `'var x` does not.

 Who should set `$implicit` is a good question, but the previous code had it this way, so I did not change it.

**Comment by vicb** [comment]
LGTM after the duplicated test is updated (see inline comment `var` -> `#`)

**Comment by tbosch** [comment]
@mhevery could you rebase?

**Comment by vicb** [comment]
removing merge as unit tests need updates

File: modules/angular2/test/change_detection/parser/parser_spec.js (hunk 2 above)
Comment (mhevery) [inline modules/angular2/test/change_detection/parser/parser_spec.js]: fixed

**Comment by vicb** [comment]
LGTM

**Pull request closed without merging**